src/UserSettingsPage.cpp

```diff
@@ -759,745 +761,858 @@ UserSettings::save()
+            } else
+                return false;
+        }
+        case DesktopNotifications: {
+            if (value.userType() == QMetaType::Bool) {
+                i->setDesktopNotifications(value.toBool());
+                return true;
+            } else
+                return false;
+        }
+        case AlertOnNotification: {
+            if (value.userType() == QMetaType::Bool) {
+                i->setAlertOnNotification(value.toBool());
+                return true;
+            } else
+                return false;
+        }
+        case AvatarCircles: {
+            if (value.userType() == QMetaType::Bool) {
+                i->setAvatarCircles(value.toBool());
+                return true;
+            } else
+                return false;
+        }
+        case UseIdenticon: {
+            if (value.userType() == QMetaType::Bool) {
+                i->setUseIdenticon(value.toBool());
+                return true;
+            } else
+                return false;
+        }
+        case DecryptSidebar: {
+            if (value.userType() == QMetaType::Bool) {
+                i->setDecryptSidebar(value.toBool());
+                return true;
+            } else
+                return false;
+        }
+            return i->decryptSidebar();
+        case PrivacyScreen: {
+            if (value.userType() == QMetaType::Bool) {
+                i->setPrivacyScreen(value.toBool());
+                return true;
+            } else
+                return false;
+        }
+        case PrivacyScreenTimeout: {
+            if (value.userType() == QMetaType::Int) {
+                i->setPrivacyScreenTimeout(value.toInt());
+                return true;
+            } else
+                return false;
+        }
+        case MobileMode: {
+            if (value.userType() == QMetaType::Bool) {
+                i->setMobileMode(value.toBool());
+                return true;
+            } else
+                return false;
+        }
+        case FontSize: {
+            if (value.userType() == QMetaType::Double) {
+                i->setFontSize(value.toBool());
+                return true;
+            } else
+                return false;
+        }
+        case Font: {
+            if (value.userType() == QMetaType::Int) {
+                i->setFontFamily(fontDb.families().at(value.toInt()));
+                return true;
+            } else
+                return false;
+        }
+        case EmojiFont: {
+            if (value.userType() == QMetaType::QString) {
+                i->setFontFamily(
+                  fontDb.families(QFontDatabase::WritingSystem::Symbol).at(value.toInt()));
+                return true;
+            } else
+                return false;
+        }
+        case Ringtone: {
+            if (value.userType() == QMetaType::Int) {
+                int ringtone = value.toInt();
+
+                // setRingtone is called twice, because updating the list breaks the set value,
```

Comment:
Surely this is some kind of bug?

Reply:
Well... it has to do with this:

The Options list is bound to the values and value property. But there is no order in which you can update them, without it being wrong.

- If you update values first, your value will point at the wrong element and cause a setData, because the index changed. So now the wrong value will be stored.
- If you update the value first, it will not be a valid index, if it points to the 4th element now. So it will be set to 0. Which again is wrong.

We can sidestep that by just updating it twice.